api/apikeys.go

```diff
@@ -0,0 +1,69 @@
+package api
+
+import (
+	"net/http"
+
+	"github.com/SkynetLabs/skynet-accounts/jwt"
+	"github.com/julienschmidt/httprouter"
+)
+
+// userAPIKeyPOST creates a new API key for the user.
+func (api *API) userAPIKeyPOST(w http.ResponseWriter, req *http.Request, _ httprouter.Params) {
+	sub, _, _, err := jwt.TokenFromContext(req.Context())
```

Comment:
nit: seems like we also have this and don't use the `sub` outside of getting the user. Could refactor to a helper.

Reply:
PJ proposed below to combine with with the `validate` (now `withAuth`) middleware and embed the user var in the context instead of the JWT token. I think that's a great solution for this.